### ISSUE TITLE
Android 10 GApps has permission problems when installing some applications/games with OBBs

### ISSUE THREAD

**Issue by KingWaffleIII** [casualsnek/waydroid_script#31]
Due to a new feature in Android 11, applications must request full file permissions to install apps/games with OBBs. However, Play Store, which installed with the script, is intended for Android 10 and thus, doesn't request these new permissions. This version of Play Store works fine for applications without OBBs but when it tries to install apps that do, the download stops mid-way and reports "Can't download  [APPNAME]":

![image](https://user-images.githubusercontent.com/43790097/150652287-44ae622f-5bc2-4fa1-a310-4bc9a8bf3c14.png)

Android 11 GApps are required to install these applications. The following image shows that on Android 11, there is separate permission for media and full file access. Google Play Store doesn't have access to the latter.

![image](https://user-images.githubusercontent.com/43790097/150652358-90fe4280-afb5-4c5d-b88f-a44e4cd1ca26.png)

**Comment by KingWaffleIII** [comment]
I think it would be best to have sources for both versions since Android 10 is still the official image given by Waydroid. Perhaps you could specify the needed version with a flag?

`sudo python waydroid_script.py -g --android-ver 11`?

**Comment by casualsnek** [comment]
We can automatically detect if image is android 10 or 11 by checking the init.rc path, or reading props :)

**Comment by herbertstew** [comment]
It works for me on my android 10 phone using aurora store to install and update games with OBB. Have you tried using that?

**Comment by KingWaffleIII** [comment]
> It works for me on my android 10 phone using aurora store to install and update games with OBB. Have you tried using that?

Android 10 should work fine in theory. The problem is Android 10 does not support ARM translation to x86, but Android 11 does. And Android 11 has this permission issue.

**Comment by KingWaffleIII** [comment]
Any updates on this?

**Comment by casualsnek** [comment]
> Any updates on this?

Not really, i was (am) quite busy on my college stuff for a few days, maybe i will work on it after ~2days !

**Comment by KingWaffleIII** [comment]
> > Any updates on this?
> 
> Not really, i was (am) quite busy on my college stuff for a few days, maybe i will work on it after ~2days !

Alright, it's no problem. Please take your time - I apologise for rushing you.

**Comment by aayushman0** [comment]
I am not sure if this will work for everyone but here is how I got it to wok:
- goto ~/.local/share/waydorid/data/
- open terminal there
- chmod -R 777 media
- lauch waydroid and launch the app

I tried it in Arknights and it worked
